### ISSUE TITLE
Collapse show/hide/toggle methods don't work while a previous animation is still running

### ISSUE THREAD

**Issue by MiladSadinam** [twbs/bootstrap#16741]
Using 

```javascript
$('.collapse').collapse('show');
$('.collapse').collapse('hide');
```
ends up in state 'show'. The second `collapse()` is being ignored because the first animation is still running.

https://jsfiddle.net/fjb5r9s5/2/ demonstrates that clicking <b>quickly</b> the button results in wrong states.

I could reproduce this in the current versions of Chrome, Firefox and IE.

EDIT: Fixed HTML5 Validator error

**Comment by twbs-lmvtfy** [comment]
Hi @MiladSadinam!

You appear to have posted a live example (https://fiddle.jshell.net/fjb5r9s5/1/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 49, column 57 thru line 49, column 64: Stray end tag ```` input ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
Yes, these methods are async. You need to use [the events they fire](http://getbootstrap.com/javascript/#collapse-events) to wait for completion.
This should be documented better, like we do for the analogous tooltip/popover methods.

**Comment by MiladSadinam** [comment]
I see what you mean. In case of my jsfiddle example I see two solutions:
- On button click I disable the button for the time until "shown.bs.collapse"/"hidden.bs.collapse" is triggered
- I manually recall `collapse()` on "shown.bs.collapse"/"hidden.bs.collapse" when the current collapse state is "wrong".

Both solutions are not optimal. Perhaps there is a better solution? Like any `collapse()` call overrides previous animations?

**Comment by cvrebert** [comment]
Or I suppose we could add some logic similar to https://github.com/twbs/bootstrap/blob/92276315fb1846ad3361a6e1e3b7a278d54fba4e/js/carousel.js#L88 to postpone the reaction to a click when (un)collapsing is already in progress.

**Comment by MiladSadinam** [comment]
Yeah, this makes the method definitely more intuitive. Thanks for the quick responses.

**Comment by mdo** [comment]
[**Bootstrap 3 is no longer being officially developed or supported.**](https://github.com/twbs/bootstrap/issues/20631)

All work has moved onto our next major release, v4. As such, this issue or pull request is being closed as a "won't fix." For additional help and support, we recommend [utilizing our community resources](https://github.com/twbs/bootstrap/blob/master/README.md#community). Thanks for your understanding, and see you on the other side of v4!

<3,
@mdo and team
